test/unit/math/prim/scal/fun/choose_test.cpp

```diff
@@ -0,0 +1,35 @@
+#include <stan/math/prim/scal.hpp>
+#include <boost/math/special_functions/fpclassify.hpp>
+#include <gtest/gtest.h>
+
+template <typename T_N, typename T_n>
+void test_choose(const T_N& N, const T_n& n) {
+  using stan::math::choose;
+  EXPECT_FLOAT_EQ(std::exp(lgamma(N + 1) - lgamma(n + 1) - lgamma(N - n + 1)),
+                  choose(N, n));
+}
```

Comment:
I want to see the test done without rounding using lgamma, because that'll match the potential rounding errors of the implementation.